src/ripple/app/ledger/tests/Ledger_test.cpp

```diff
@@ -0,0 +1,149 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/tests/common_ledger.h>
+
+namespace ripple {
+namespace test {
+
+class Ledger_test : public beast::unit_test::suite
+{
+    void test_genesisLedger (bool sign)
+    {
+        std::uint64_t const xrp = std::mega::num;
+
+        auto master = createAccount ("masterpassphrase");
+
+        Ledger::pointer LCL = createGenesisLedger(100000*xrp, master);
+
+        Ledger::pointer ledger = std::make_shared<Ledger>(false, *LCL);
+
+        // User accounts
+        auto gw1 = createAccount ("gw1");
+        expect (gw1.pk != master.pk, "gw1.pk != master.pk");
```

Comment:
Tom added an expectEquals, which, provided the values are `<<` stream compatible, prints out each value upon failure. You can argue that it doesn't matter in this case, and all that's really of interest is that it `!=` and `gw1.pk != master.pk` is sufficient information. Which brings me nicely to my next point: when you see `#10 gw1.pk != master.pk` what's the next step? You need to hunt down the location in the source right?

We should consider adding some macros like EXPECT_EQ that delegate to Tom's expectEquals method. We would get:

1.  The expect expression for free, with no need to repeat again as a string
2.  The expected and actual values for free
3.  The file / line position of the error
4.  Free uniform semantics of the error message
   1. 1 person might use the second arg to expect to say it failed because $msg
   2. Another might use it to repeat their expectation (as here)

The more clues you get, the quicker you can debug something, so we should make clues cheaper. I've seen quite a few instances of expect($expr, $do-stuff-to-build-up-some-message-we-should-just-have) in our tests, as well as explicit try/catch blocks that could be handled by ripple::TestSuite::expectException.

Which brings me to my next point. What's the purpose of this suite of utilities/tests? Where is it going? Do we want this to evolve into an API that people can use to write tests WHILE they are developing in C++? Or is this just a simple exercise of some ledger related code? 

I note that applyTransaction assumes tesSUCCESS, but most of the interesting tests will be the paths less beaten (tem, tec, tef, etc)